### PR TITLE
Fix various issues with log format specifiers not matching the types being logged

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
#### Problem
I was looking at our logs and saw things like:
```
I (174870) chip[EM]: Received message of type 8 and protocolId 327680 on exchange 44249
```
and
```
I (141230) chip[SVR]: Packet received from Node:3ffe5f20: 112233 bytes 0
```

which were obviously bogus-looking.  The protocol id there is `0x00050000` when it should be `0x5` and we did not get a message that's `112233` bytes long, obviously.

Turns out that first is because we were logging a struct as an int, and the second is because we logged a 64-bit node id as a 32-bit quantity and the remaining 4 bytes leaked over into the other specifiers.

#### Change overview
1. Enable warnings, at least on gcc and clang, for chip::Logging::Log formats not matching what's being logged.
2. Fix a bunch of the resulting build errors.
3. Rerun codegen.

#### Testing
1. Made sure this compiled on Mac, ESP32, nrfconnect.
2. Checked that the bogus logs from above now look like this:
    ```
    I (71890) chip[EM]: Received message of type 8 and protocolId 5 on exchange 44249
    ```
    and
    ```
    I (103520) chip[SVR]: Packet received from Node 0x000000000001B669: 16 bytes
    ```

### CODE REVIEW

**Review by sagar-apple — APPROVED** [review]

**Review by tcarmelveilleux — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: src/platform/OpenThread/OpenThreadUtils.cpp [hunk 1]
```diff
@@ -100,7 +100,7 @@ void LogOpenThreadStateChange(otInstance * otInst, uint32_t flags)
 
     static char strBuf[64];
 
-    ChipLogDetail(DeviceLayer, "OpenThread State Changed (Flags: 0x%08x)", flags);
+    ChipLogDetail(DeviceLayer, "OpenThread State Changed (Flags: 0x%08" PRIu32 ")", flags);
```
Comment (tima-q) [inline src/platform/OpenThread/OpenThreadUtils.cpp:103]: Given the 0x - should this flag print be a PRIx32 iso PRIu32 ?

Comment (bzbarsky-apple) [reply src/platform/OpenThread/OpenThreadUtils.cpp:103]: Yes, good catch!

**Comment by bzbarsky-apple** [comment]
@andy31415 @msandstedt @saurabhst @Damian-Nordic @jmartinez-silabs could you take a look, please?

File: src/transport/AdminPairingTable.cpp [hunk 2]
```diff
@@ -352,7 +352,8 @@ AdminPairingInfo * AdminPairingTable::FindAdminForNode(FabricId fabricId, NodeId
         if (state.IsInitialized())
         {
             ChipLogProgress(Discovery,
-                            "Checking ind:%d [fabricId 0x" ChipLogFormatX64 " nodeId 0x" ChipLogFormatX64 " vendorId %d] vs"
+                            "Checking ind:%" PRIu32 " [fabricId 0x" ChipLogFormatX64 " nodeId 0x" ChipLogFormatX64
+                            " vendorId %d] vs"
```
Comment (Damian-Nordic) [inline src/transport/AdminPairingTable.cpp:356]: use PRIu16 for vendorId?

**Review by Damian-Nordic — APPROVED** [review]

File: src/transport/AdminPairingTable.cpp (hunk 2 above)
Comment (bzbarsky-apple) [reply src/transport/AdminPairingTable.cpp:356]: Yep, good catch.

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**